sql-plugin/src/main/java/com/nvidia/spark/rapids/fileio/hadoop/HadoopInputFile.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.fileio.hadoop;
+
+import com.nvidia.spark.rapids.fileio.RapidsInputFile;
+import com.nvidia.spark.rapids.fileio.SeekableInputStream;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class HadoopInputFile implements RapidsInputFile {
+    private final Path filePath;
+    private final FileSystem fs;
+    private volatile FileStatus fileStatus;
+
+    public static HadoopInputFile create(Path filePath, Configuration conf) throws IOException {
+        Objects.requireNonNull(filePath, "filePath can't be null!");
+        Objects.requireNonNull(conf, "Hadoop conf can't be null");
+        FileSystem fs = filePath.getFileSystem(conf);
+        return new HadoopInputFile(filePath, fs);
+    }
+
+    private HadoopInputFile(Path filePath, FileSystem fs) {
+        Objects.requireNonNull(filePath, "filePath can't be null!");
+        Objects.requireNonNull(fs, "FileSystem can't be null");
+        this.filePath = filePath;
+        this.fs = fs;
+        this.fileStatus = null;
+    }
+
+    @Override
+    public long getLength() throws IOException {
+        ensureFileStatus();
+        return fileStatus.getLen();
+    }
+
+    @Override
+    public SeekableInputStream open() throws IOException {
+        return new HadoopInputStream(fs.open(filePath));
+    }
+
+    private void ensureFileStatus() throws IOException {
+        if (fileStatus == null) {
```

Comment:
Why do we need this double checking for something that is called only once for footer parsing?

Reply:
This is not a hot call path, called only once.